enhancements/microshift/update-and-rollback.md

```diff
@@ -0,0 +1,180 @@
+my application that work in production.
+
+As an application deployer, I want to focus only on the lifecycle management for
+my application and forget about how MicroShift is upgraded.
+
+As an application deployer, I can provide new versions of my application as part
+of the OSTree, and enforce its upgrade
+
+As a device administrator, I can provide a new version of the Operating System
+OSTree and the edge device will upgrade automatically.
+
+As a device administrator, I need to ensure a MicroShift upgrade works and a
+rollback mechanism is in place in case something goes wrong.
+
+### Goals
+
+The goals of this proposal are:
+- How to implement a successful MicroShift upgrade on systems based on `rpm-ostree`
+- Describe the process of creating a MicroShift's data backup
+- Identify what files need to be saved as part of a MicroShift's data backup
+- Describe the process of rolling back to a previous working state
+
+### Non-Goals
+
+The non-goals of this proposal are:
+- How to implement MicroShift healthchecks using `greenboot`
+- How to implement upgrades of 3rd party applications running on MicroShift
+- How to implement rollbacks of 3rd party applications running on MicroShift
+
+## Proposal
+
+### Workflow Description
+
+The Operating System of an Edge Device running MicroShift will be created using
+Image Builder. This tool will allow `device administrators` to customize an edge
+optimized version of RHEL by adding RPMs and configuration of choice as described
+[here](https://github.com/openshift/enhancements/blob/master/enhancements/microshift/kubernetes-for-device-edge.md).
+
+When a new version of MicroShift needs to be rolled out, a new image shall be created
+and exposed to the edge devices. When working with thousands of Edge Devices,
+automatic detection and enforcement of upgrades is required.  Hence, Edge Devices will
+automatically detect the new exposed OSTree and perform a transactional upgrade.
+
+Before pivoting to the new Operating System Tree, a data backup needs to be performed
+in order to save the state of the running Edge Device. Due to the resource constrained
+environments where these Edge Devices run on, only the N-1 state will be supported. That 
+means that only one backup will be stored and when the upgrade is performed, the Edge
+Device can only rollback to the previous state N-1.
+
+The backup should consider all data required to save the state of MicroShift such as
+certificates, configuration files, volumes
+
+At every boot, `Greenboot` healthchecks will be executed as described in [this enhancement.](https://github.com/openshift/enhancements/blob/master/enhancements/microshift/microshift-greenboot.md)
+
+If upgrading MicroShift to a new version results in a broken state, `Greenboot` scripts
+will restore the previously saved backup, and perform an atomic rollback to the N-1 state.
+
+### API Extensions
+
+None
+
+### Implementation Details/Notes/Constraints [Optional]
+
+#### rpm-ostree Configuration
```

Comment:
No, this is a pre-requisite to enforce rpm-ostree upgrades in an automatic manner. Otherwise, the administrator of the device (or thousands of devices) will have to execute commands to perform the upgrade manually.

I agree it should be more described in a requirement section, or side note to the overall MicroShift upgrade process.

Reply:
OK, just saying something like that here in the document will make it clear that this is more background material.